src/users/controllers/user.controller.ts

```
@@ -0,0 +1,22 @@
+import { Controller, Inject, Post, Body } from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { UserDiTokens } from '../di/user-tokens.di';
+import { RegisterUserService } from '../services/register-user.service';
+import { RegisterUserPort } from '../services/usecases/register-user.usecase';
+
+@ApiTags('Users')
+@Controller('users')
+export class UserController {
+  constructor(
+    @Inject(UserDiTokens.RegisterUserService)
+    private readonly registerUserService: RegisterUserService,
+  ) {}
+
+  @Post('register')
+  @ApiOperation({ summary: 'Register a new user' })
+  @ApiResponse({ status: 201, description: 'User registered successfully' })
+  async register(@Body() payload: RegisterUserPort) {
+    const user = await this.registerUserService.execute(payload);
+    return { message: 'User registered successfully!', user };
```

you should not return the full user since you would be exposing user password and id to the client. You should create a new data type (dto - data transfer object), which would have all of the same properties as user object except password and id. Put the this dto in a seperate directory called "dto".